dom0-updates/fwupd-dom0-update

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+escape_args() {
+    local eargs=""
+
+    for arg in "$@"; do
+    printf -v eargs '%s%q ' "$eargs" "$arg"
+    done
+
+    echo "${eargs%?}"
+}
+
+find_regex_in_args() {
+    local regex="${1}"
+    shift 1
+
+    for arg in "${@}"; do
+        if echo "${arg}" | grep -q -e "${regex}"; then
+            return 0
+        fi
+    done
+
+    return 1
+}
+
+UPDATEVM=`qubes-prefs --force-root updatevm`
+FWUPD_DOM0_CACHE_FILES=/root/.cache/fwupd
+FWUPD_UPDATEVM_CACHE_FILES=/home/user/.cache/fwupd
+
+#if [ -z "$UPDATEVM" ]; then
+#    echo "UpdateVM not set, exiting"
+#    exit 1
+#fi
+
+if [ "$1" = "--help" ]; then
+    echo "This tool is used to download packages for dom0. Without package list"
+    echo "it checks for updates for installed packages"
+    echo ""
+    echo "Usage: $0 [--clean] [--check-only] [--gui] [<pkg list>]"
+    echo "    --clean      clean cache before doing anything"
+    echo "    --metadata   inits metadata download"
+    echo "    --update     inits firmware update files download"
+    echo "    --url=       firmware update url"
+    echo "    --sha=       firmware upadte sha1 checksum"
+    exit
+fi
+
+CLEAN=
+METADATA=
+UPDATE=
+URL=
+SHASUM=
+FW_NAME=
+
+# Filter out some dnf options and collect packages list
+while [ $# -gt 0 ]; do
+    case "$1" in
+        --clean)
+            CLEAN=1
+            ;;
+        --metadata)
+            METADATA=1
+            ;;
+        --update)
+            UPDATE=1
+            ;;
+        --url=*)
+            URL=${1#--url=}
+            FW_NAME=${1#--url=https://fwupd.org/downloads/}
+            ;;
+        --sha=*)
+            SHASUM=${1#--sha=}
+            ;;
+        *)
+            echo "Command unknown"
```

Comment:
`Command unknown` sounds weird to me, i would swap those. I also like to print the `unknown` command to the user.